components/kyma-environment-broker/internal/process/provisioning/cls_binding_step.go

```diff
@@ -0,0 +1,171 @@
+	}
+	smCli := operation.SMClientFactory.ForCredentials(smCredentials)
+
+	// test if the provisioning is finished, if not, retry after 10s
+	resp, err := smCli.LastInstanceOperation(operation.Cls.Instance.InstanceKey(), "")
+	if err != nil {
+		failureReason := fmt.Sprintf("LastInstanceOperation() call failed")
+		log.Errorf("%s: %s", failureReason, err)
+		return s.operationManager.OperationFailed(operation, failureReason)
+	}
+	log.Infof("Provisioning Cls (instanceID=%s) state: %s", operation.Cls.Instance.InstanceID, resp.State)
+	switch resp.State {
+	case servicemanager.InProgress:
+		return operation, 10 * time.Second, nil
+	case servicemanager.Failed:
+		failureReason := fmt.Sprintf("Cls instance is state failed")
+		log.Errorf("%s: %s", failureReason, resp.Description)
+		return s.operationManager.OperationFailed(operation, fmt.Sprintf("Cls provisioning failed: %s", resp.Description))
+	case servicemanager.Succeeded:
+		operation.Cls.Instance.Provisioned = true
+		operation.Cls.Instance.ProvisioningTriggered = false
+		log.Info("Cls instance is provisioned.")
+	}
+
+	var overrides *cls.ClsOverrideParams
+
+	if !operation.Cls.Binding.Bound {
+		if operation.Cls.Binding.BindingID == "" {
+			operation.Cls.Binding.BindingID = uuid.New().String()
+		}
+
+		// Create a binding
+		overrides, err = s.bindingProvider.CreateBinding(smCli, &cls.BindingRequest{
+			InstanceKey: operation.Cls.Instance.InstanceKey(),
+			BindingID:   operation.Cls.Binding.BindingID,
+		})
+
+		if err != nil {
+			failureReason := fmt.Sprintf("Cls Binding failed")
+			log.Errorf("%s: %s", failureReason, err)
+			return s.operationManager.OperationFailed(operation, failureReason)
+		}
+
+		encryptedOverrides, err := cls.EncryptOverrides(s.secretKey, overrides)
+		if err != nil {
+			failureReason := fmt.Sprintf("encryptClsOverrides() call failed")
+			log.Errorf("%s: %s", failureReason, err)
+			return s.operationManager.OperationFailed(operation, failureReason)
+		}
+
+		operation.Cls.Overrides = encryptedOverrides
+		operation.Cls.Binding.Bound = true
+
+		// save the status
+		op, retry := s.operationManager.UpdateOperation(operation)
+		if retry > 0 {
+			log.Errorf("unable to update operation")
+			return operation, time.Second, nil
+		}
+		operation = op
+	} else {
+		// fetch existing overrides
+		overrides, err = cls.DecryptOverrides(s.secretKey, operation.Cls.Overrides)
+		if err != nil {
+			failureReason := fmt.Sprintf("decryptClsOverrides() call failed")
+			log.Errorf("%s: %s", failureReason, err)
+			return s.operationManager.OperationFailed(operation, failureReason)
+		}
+	}
+
+	operation.InputCreator.SetLabel(kibanaURLLabelKey, overrides.KibanaUrl)
+	flOverride, err := s.injectOverrides(overrides, log)
+	if err != nil {
+		log.Errorf("Unable to generate forward plugin to push logs: %v", err)
+		return operation, time.Second, nil
+	}
+
+	checkKymaVersion, err := cls.IsKymaVersion_1_20(operation.RuntimeVersion.Version)
```

Comment:
checkKymaVersion does not say what does it mean, maybe isVersion1_20

Reply:
fixed.